pkg/scanners/scannerv4/convert.go

```diff
@@ -42,6 +42,7 @@ func components(metadata *storage.ImageMetadata, report *v4.VulnerabilityReport)
 			Name:          pkg.GetName(),
 			Version:       pkg.GetVersion(),
 			Vulns:         vulnerabilities(report.GetVulnerabilities(), vulnIDs),
+			Source:        sourceType(pkg),
 			Location:      pkg.GetPackageDb(),
```

Comment:
I originally didn't do that because I wasn't sure if we could assume `:` is an invalid character in a filename. I added it and commented the assumption that it's invalid

Reply:
Perhaps can leverage the `sourceType` logic in both cases - ie: if the data before the `:` is a valid source type, then strip it from the location, otherwise leave as is.

Reply:
updated [99dd2c4](https://github.com/stackrox/stackrox/pull/9444/commits/99dd2c4113f299da6108d2610827cf7f6c176727)